AuroraPatch/Program.cs

```diff
@@ -73,26 +82,37 @@ private static Form GetTacticalMap(Assembly assembly)
 
                     if (buttons >= 60 && buttons <= 80 && checkboxes >= 60 && checkboxes <= 80)
                     {
-                        Debug.WriteLine("Map: " + type.Name);
+                        logger.LogDebug("TacticalMap found: " + type.Name);
                         var map = (Form)Activator.CreateInstance(type);
 
                         return map;
                     }
                 }
             }
 
-            throw new Exception("Tactical Map not found");
+            logger.LogCritical("TacticalMap not found");
+            // If we expose more forms/functionality in the future, may want to make this an error instead
```

Comment:
I think you misunderstood why the TacticalMap is being searched for, it's not about exposing the form as such to allow patches to interface with it. Everything is already exposed through the `TacticalMap..GetType().Assembly` property, see the Harmony example in the ExamplePatch project, where the `InitializeComponent` function of the Economics form is being patched. The idea is that patch authors can use the `Program.AuroraChecksum` property to check what Aurora version they are patching, and based on that decide to use whatever method they want for patching the Assembly, which is reachable through `TacticalMap..GetType().Assembly`. The reason we need to find the TacticalMap is so as to start the game, because we're loading the Aurora exe as a library we have to replicate the code to actually start the game, meaning doing a call to `Application.Run(map)` where the `map` parameter is an instance of the TacticalMap.